### PR TITLE
Add Player Attack and Basic Spell

### PR DESCRIPTION
**Pull request by daniela-venuta**

### CODE REVIEW

File: Hush/Assets/Scripts/Weapon/PlayerWeaponController.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        equippedWeapon.PerformSpecialAttack();
+    }
+
+    private int CalculateDamage()
+    {
+        Stat.StatType type = (equippedWeapon == sword) ? Stat.StatType.Strength : Stat.StatType.SpellPower;
+        int damageToDeal = (characterStats.GetStat(type).GetCalculatedStatValue() * 2);
+        damageToDeal += CalculateCrit(damageToDeal);
+        
+        //TODO damage ui?
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Weapon/PlayerWeaponController.cs:55]: yes good call. I'll make a task for UI

File: Hush/Assets/Scripts/Weapon/Spell.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+{
+    [SerializeField] private Animator animator;
+    public int CurrentDamage { get; set; }
+    public int BonusDamage { get; set; }
+
+    private const int SPECIAL_DAMAGE = 50;
+
+    void Awake()
+    {
+        BonusDamage = 5;
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Weapon/Spell.cs:16]: ### Minor tweak
Could be better to separate the property & its getter/setter that way you can auto assign it a value. by default BUT also have the value from the inspector get assigned should you decide to overwrite it in the inspector. 
ie:
``` C#
[SerializeField] private int bonusDamage = 5; // actual property
public int BonusDamage 
{
    get => bonusDamage; 
    set => bonusDamage = value;
 }
```

**Review by arejayelle — COMMENTED** [review]
Overall good stuff &  works great!
Just a few small comments to be addressed and then I'll approve.

File: Hush/Assets/Scripts/Player/StatBonus.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class StatBonus
```
[the remark this answers is not in the document]
Comment (daniela-venuta) [reply Hush/Assets/Scripts/Player/StatBonus.cs:5]: deleted it and kept a list of int

File: Hush/Assets/Scripts/Weapon/Spell.cs (hunk 2 above)
Comment (daniela-venuta) [reply Hush/Assets/Scripts/Weapon/Spell.cs:16]: sounds good!

File: Hush/Assets/Scripts/Player/Enums/StatType.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Enums
+{
+    public enum StatType { Strength , SpellPower }
+}
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Player/Enums/StatType.cs:8]: Thanks <3

File: Hush/Assets/Scripts/Player/Stat.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Enums;
+
+public class Stat
+{
+    //public enum StatType { Strength , SpellPower }
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Player/Stat.cs:8]: Remove line

```suggestion

public class Stat
{
```

File: Hush/Assets/Scripts/Weapon/PlayerWeaponController.cs (hunk 1 above)
Comment (arejayelle) [reply Hush/Assets/Scripts/Weapon/PlayerWeaponController.cs:55]: Created task #44

**Review by arejayelle — CHANGES_REQUESTED** [review]
Last bit of cleanup I promise!

**Review by arejayelle — APPROVED** [review]
All good!

**Pull request merged**